### ISSUE TITLE
'Dataset' object has no attribute 'drop_vars'

### ISSUE THREAD

**Issue by fluidnumerics-joe** [MITgcm/xmitgcm#292]
I'm trying to load in a dataset that is using sphericalpolar coordinates, but am running into issues in the `_swap_dimensions` method.

Below is a snippet of code I'm using to load the dataset
```
#!/usr/bin/env python3


from xmitgcm import open_mdsdataset


dataDir = '/tank/tdgs/simulations/MITGCM300_z150/design2/run/spinup/metadata'

ds = open_mdsdataset(data_dir = dataDir,
                     grid_dir = dataDir,
                     iters = 'all',
                     prefix = ['U','V','W','T','S'],
                     read_grid = True,
                     delta_t = 120,
                     ref_date = "2013-1-1 0:0:0",
                     geometry = "sphericalpolar")
```

And below is the backtrace I'm seeing.
```
Traceback (most recent call last):
  File "./xmitgcm_demo.py", line 16, in <module>
    geometry = "sphericalpolar")
  File "/tank/users/schoonover/topocontrol-gulfstream/post-process/xmitgcm/env/lib/python3.5/site-packages/xmitgcm/mds_store.py", line 244, in open_mdsdataset
    ds = _swap_dimensions(ds, geometry)
  File "/tank/users/schoonover/topocontrol-gulfstream/post-process/xmitgcm/env/lib/python3.5/site-packages/xmitgcm/mds_store.py", line 325, in _swap_dimensions
    ds = ds.drop_vars(orig_dim)
  File "/tank/users/schoonover/topocontrol-gulfstream/post-process/xmitgcm/env/lib/python3.5/site-packages/xarray/core/common.py", line 230, in __getattr__
    "%r object has no attribute %r" % (type(self).__name__, name)
AttributeError: 'Dataset' object has no attribute 'drop_vars'
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7f2a120690d0>
Traceback (most recent call last):
  File "/usr/lib/python3.5/weakref.py", line 117, in remove
TypeError: 'NoneType' object is not callable
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7f2a120690d0>
Traceback (most recent call last):
  File "/usr/lib/python3.5/weakref.py", line 117, in remove
TypeError: 'NoneType' object is not callable
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7f2a120690d0>
Traceback (most recent call last):
  File "/usr/lib/python3.5/weakref.py", line 117, in remove
TypeError: 'NoneType' object is not callable
****
```
The relevant line of code where the error occurs is [here](https://github.com/MITgcm/xmitgcm/blob/master/xmitgcm/mds_store.py#L350)

For reference, I am using Python 3.5.3 and the following packages in a virtual environment
```
cachetools==4.2.4
dask==2.6.0
docrep==0.3.2
future==0.18.2
numpy==1.18.5
pandas==0.25.3
pkg-resources==0.0.0
python-dateutil==2.8.2
pytz==2021.3
six==1.16.0
toolz==0.11.2
xarray==0.13.0
xgcm==0.3.0
xmitgcm==0.5.0
```

**Comment by timothyas** [comment]
Hey there! I believe this is an xarray related issue. It looks like `.drop_vars` was removed in [xarray version 0.9.0](https://xarray.pydata.org/en/stable/whats-new.html#v0-9-0-25-january-2017) and then brought back in [xarray version 0.14.1](https://xarray.pydata.org/en/stable/whats-new.html#v0-14-1-19-nov-2019). It looks like you're in the "sweet" spot of versions where that method doesn't exist. Could you try updating and see if that fixes the issue?

**Comment by fluidnumerics-joe** [comment]
Hey @timothyas , thanks for the suggestion  - upgrading xarray indeed solved the problem.

**Comment by timothyas** [comment]
Closed via #293
